azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1796,17 +1806,13 @@ private Permission getPermissionObject(final Project project, final User user,
     return perm;
   }
 
-  private boolean hasPermissionToCreateProject(final User user) {
-    for (final String roleName : user.getRoles()) {
-      final Role role = this.userManager.getRole(roleName);
-      final Permission perm = role.getPermission();
-      if (perm.isPermissionSet(Permission.Type.ADMIN)
-          || perm.isPermissionSet(Permission.Type.CREATEPROJECTS)) {
-        return true;
-      }
-    }
+  private void canUploadProjects(final Page page, final Session session) {
```

Comment:
The name seems to suggest that it will return a true or false as a check.

Can you come up a better name? e.g. hideUplodButtonWhenNeeded?

It doesn't look like there is test that covers this method.